rust/kernel/endian.rs

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Endian integer types
+
+macro_rules! define_le_integer {
+    ($name:ident, $native_type:ty) => {
+        /// little-endian integer
+        #[allow(non_camel_case_types)]
+        #[derive(Default, Clone, Copy)]
+        #[repr(transparent)]
+        pub struct $name($native_type);
+
+        impl From<$native_type> for $name {
+            #[inline(always)]
+            fn from(v: $native_type) -> Self {
+                Self(<$native_type>::from_le(v))
+            }
+        }
+
+        impl From<$name> for $native_type {
+            #[inline(always)]
+            fn from(v: $name) -> Self {
+                <$native_type>::from_le(v.0)
+            }
+        }
+    };
+}
+
+macro_rules! define_be_integer {
+    ($name:ident, $native_type:ty) => {
+        /// big-endian integer
+        #[allow(non_camel_case_types)]
+        #[derive(Default, Clone, Copy)]
+        #[repr(transparent)]
+        pub struct $name($native_type);
+
+        impl From<$native_type> for $name {
+            #[inline(always)]
+            fn from(v: $native_type) -> Self {
+                Self(<$native_type>::from_be(v))
+            }
+        }
+
+        impl From<$name> for $native_type {
+            #[inline(always)]
+            fn from(v: $name) -> Self {
+                <$native_type>::from_be(v.0)
+            }
+        }
+    };
+}
+
+define_le_integer!(le64, u64);
+define_le_integer!(le32, u32);
+define_le_integer!(le16, u16);
+
+define_be_integer!(be64, u64);
+define_be_integer!(be32, u32);
+define_be_integer!(be16, u16);
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn check_size() {
+        assert_eq!(core::mem::size_of::<be16>(), core::mem::size_of::<u16>());
+        assert_eq!(core::mem::size_of::<be32>(), core::mem::size_of::<u32>());
+        assert_eq!(core::mem::size_of::<be64>(), core::mem::size_of::<u64>());
+        assert_eq!(core::mem::size_of::<le16>(), core::mem::size_of::<u16>());
+        assert_eq!(core::mem::size_of::<le32>(), core::mem::size_of::<u32>());
+        assert_eq!(core::mem::size_of::<le64>(), core::mem::size_of::<u64>());
```

Comment:
Could these be moved into the `define*!` macros and made `static_assert!`/`build_assert!`?

Reply:
Done.